server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsResponse.java

```diff
@@ -88,12 +90,42 @@ public ClusterStatsResponse(
         List<ClusterStatsNodeResponse> nodes,
         List<FailedNodeException> failures,
         ClusterState state
+    ) {
+        this(
+            timestamp,
+            clusterUUID,
+            clusterName,
+            nodes,
+            failures,
+            state,
+            ClusterStatsRequest.Metric.allMetrics(),
+            ClusterStatsRequest.IndexMetrics.allIndicesMetrics()
+        );
+    }
+
+    public ClusterStatsResponse(
+        long timestamp,
+        String clusterUUID,
+        ClusterName clusterName,
+        List<ClusterStatsNodeResponse> nodes,
+        List<FailedNodeException> failures,
+        ClusterState state,
+        Set<String> requestedMetrics,
+        Set<String> indicesMetrics
     ) {
         super(clusterName, nodes, failures);
         this.clusterUUID = clusterUUID;
         this.timestamp = timestamp;
-        nodesStats = new ClusterStatsNodes(nodes);
-        indicesStats = new ClusterStatsIndices(nodes, MappingStats.of(state), AnalysisStats.of(state));
+        nodesStats = !requestedMetrics.isEmpty() && !requestedMetrics.equals(Set.of(Metric.INDICES.metricName()))
```

Comment:
Is there an assumption that anything other than `INDICES` is nodeStats ? 
If so, can we return empty ClusterStatsNodes instead of null ?

Reply:
>Is there an assumption that anything other than INDICES is nodeStats ?
yes.

> If so, can we return empty ClusterStatsNodes instead of null ?
I think its okay to return null otherwise we will need to create a non-parameterized constructor for ClusterStatsNodes, metrics classes.

Reply:
If we add new metrics in future that is not related to Nodes, this assumption will break.

Reply:
Addressed this by maintaining a set of metrics of node stats.